File: backend/schemas/maintenance.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional, List


class MaintenanceCreate(BaseModel):
    name: str
    description: Optional[str] = None
    starts_at: datetime
    ends_at: datetime
    is_recurring: bool = False
    recurrence_cron: Optional[str] = None
    monitor_ids: Optional[List[int]] = None  # monitors to pause during window


class MaintenanceUpdate(BaseModel):
    name: Optional[str] = None
    description: Optional[str] = None
    starts_at: Optional[datetime] = None
    ends_at: Optional[datetime] = None
    is_recurring: Optional[bool] = None
    recurrence_cron: Optional[str] = None
    monitor_ids: Optional[List[int]] = None


class MaintenanceOut(BaseModel):
    id: int
    user_id: int
    name: str
    description: Optional[str]
    starts_at: datetime
    ends_at: datetime
    is_recurring: bool
    recurrence_cron: Optional[str]
    affected_monitors: Optional[str]
    created_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/api/maintenance.ts
import client from './client'

export interface MaintenanceCreate {
  name: string; description?: string; starts_at: string; ends_at: string
  is_recurring?: boolean; recurrence_cron?: string; monitor_ids?: number[]
}
export interface MaintenanceOut {
  id: number; user_id: number; name: string; description?: string
  starts_at: string; ends_at: string; is_recurring: boolean
  recurrence_cron?: string; affected_monitors?: string; created_at: string
}

export const maintenanceApi = {
  list: () => client.get<MaintenanceOut[]>('/maintenance'),
  create: (data: MaintenanceCreate) => client.post<MaintenanceOut>('/maintenance', data),
  get: (id: number) => client.get<MaintenanceOut>(`/maintenance/${id}`),
  update: (id: number, data: Partial<MaintenanceCreate>) => client.put<MaintenanceOut>(`/maintenance/${id}`, data),
  delete: (id: number) => client.delete(`/maintenance/${id}`),
  active: () => client.get<MaintenanceOut[]>('/maintenance/active'),
}
